bindings/java/src/main/java/org/openucx/jucx/examples/UcxReadBWBenchmarkSender.java

```diff
@@ -49,22 +55,23 @@ public static void main(String[] args) throws Exception {
 
         // Send memory metadata and wait until receiver will finish benchmark.
         endpoint.sendTaggedNonBlocking(sendData, null);
-        ByteBuffer recvBuffer = ByteBuffer.allocateDirect(4096);
-        UcpRequest recvRequest = worker.recvTaggedNonBlocking(recvBuffer,
-            new UcxCallback() {
-                @Override
-                public void onSuccess(UcpRequest request) {
-                    System.out.println("Received a message:");
-                    System.out.println(recvBuffer.asCharBuffer().toString().trim());
-                }
-            });
 
-        worker.progressRequest(recvRequest);
+        try {
+            while (true) {
+                while (worker.progress() == 0) {
```

Comment:
maybe like this
for (;;) {
 if (worker.progress()==0) {
   worker.waitForEvents()
 }
}